Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -2624,6 +2624,13 @@ - (void)testWatchWithMatchFilterAsync {
 
 - (NSArray<RLMObjectId *> *)insertDogDocuments:(RLMMongoCollection *)collection {
     __block NSArray<RLMObjectId *> *objectIds;
+    XCTestExpectation *ex = [self expectationWithDescription:@"delete existing documents"];
```

Comment:
No, we don't delete anything on tearDown. The sync tests use a separate partition for each test so that they don't see each others data, but these tests are accessing the collection directly.

Reply:
does deleting all documents before inserting obscure what this function is intended for? Should the calling function just call something like `deleteAllDogDocuments` and then call `insertDogDocuments`?

Reply:
We never want to call them separately, but I guess the name should be `prepareDogDocuments` or something. The thing being fixed here is that without deleting old documents the tests only work when run in a specific order and break when run in other orders due to the missing cleanup.

Reply:
Sure, lets change the name to avoid confusion in the future